source/common/ssl/ssl_socket.cc

```diff
@@ -245,7 +248,8 @@ std::string SslSocket::urlEncodedPemEncodedPeerCertificate() const {
   // URL encoding shortcut
   absl::StrReplaceAll({{"\n", "%0A"}, {" ", "%20"}, {"+", "%2B"}, {"/", "%2F"}, {"=", "%3D"}},
                       &pem);
-  return pem;
+  urlEncodedPemEncodedPeerCertificate_ = std::move(pem);
```

Comment:
I would just remove `pem` local variable and assign directly into the instance variable above.